packages/swingset-liveslots/test/test-collections.js

```diff
@@ -352,7 +352,15 @@ test('constrain set key shape', t => {
   t.deepEqual(Array.from(lt47.values(M.gt(20))), [29, 46]);
 });
 
-test('bogus key shape', t => {
+// Prior to https://github.com/Agoric/agoric-sdk/pull/7035, virtual
+// stores did a sanity check on `keyShape` that it made sense for
+// scalar keys. However, that check was both broken and not otherwise
+// needed, so #7035 removed it. Instead, for scalar collections, the
+// key will *also* be checked for being a scalar. In cases like
+// `M.promise()`, without the error check, no possible key will pass
+// both conditions, and so such stores will always be empty. But
+// we don't need to treat this as an error.
+test.failing('bogus key shape', t => {
```

Comment:
@FUDCo , is this loss of functionality ok with you? 

Note that the intended functionality never did work because of `assertKeyPattern` bugs, so this isn't loss of functionality that actually worked. For example, it would reject the `M.or(M.string(), M.promise())` below, but it would accept `M.any()` and `M.or(M.string(), M.any())`.

Reply:
This change seems fine, though I'll complain that you're violating a more general principle that it's unhelpful thus usually inappropriate for comments in code to remark upon the code's own history.  Nobody reading a piece of code cares what was there before.

Reply:
@FUDCo is certainly correct about the comment.

But in discussing this with @warner and @FUDCo , we do not have consensus on the new semantics that just intersects with `M.scalar()`. Neither do we have consensus on an error checking semantics. I believe we need a semantics more formalizable than either 
   * alert the user if it heuristically seems like they made a mistake
   * whatever the checking code actually happens to do is the semantics of the moment

Plus, I am still of the opinion that `assertKeyPattern` was not pulling its own weight. It was too much mechanism for too little benefit.